modules/obstacles_and_odometry.py

```diff
@@ -0,0 +1,52 @@
+"""
+Obstacles and local odometry merged data structure.
+"""
+
+from . import drone_odometry_local
+from . import obstacle
+
+
+class ObstaclesAndOdometry:
+    """
+    Contains obstacles and current local odometry.
+    """
+
+    __create_key = object()
+
+    @classmethod
+    def create(
+        cls,
+        obstacles: "list[obstacle.Obstacle]",
+        local_odometry: drone_odometry_local.DroneOdometryLocal,
+    ) -> "tuple[bool, ObstaclesAndOdometry | None]":
+        """
+        Combines obstacles with local odometry.
+        """
+        if len(obstacles) == 0:
+            return False, None
```

Comment:
is it actually invalid if the obstacle list is empty? what if the drone scans and just doesn't see obstacles?